plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -960,7 +960,13 @@ def create_vm(self, vmid, newid, node, name, memory, cpu, cores, sockets, update
                 kwargs.update(kwargs[k])
                 del kwargs[k]
 
-        # Rename numa_enabled to numa. According the API documentation
+        # Map bool to string, according to API documentation.
+        if kwargs['agent'] is True:
+          kwargs['agent'] = 'enabled=1'
+        elif kwargs['agent'] is False:
+          kwargs['agent'] = 'enabled=0'
```

Comment:
According to https://github.com/ansible-collections/community.general/pull/4027#discussion_r913231029 these are never true since the type is not a string. Is there a way to make the type support either a boolean or a string.